src/misc/setup.cpp

```diff
@@ -674,13 +676,36 @@ const std::vector<Value>& PropMultiVal::GetValues() const
 	return valid_values;
 }
 
+// When setting a property's list of valid values (For example, composite =
+// [auto, on, off]), this function inspects the given valid value to see if it's
+// boolean string (ie: "on" or "off"). If so, this records if a boolean is valid
+// and its direction (either positive or negative) so we can accept all of those
+// corresponding boolean strings from the user (ie: composite = disabled")
+//
+void Property::MaybeSetBoolValid(const std::string_view valid_value)
+{
+	// Deliberately ignore the numerical bools to avoid false-positive
+	if (valid_value == "0" || valid_value == "1") {
+		return;
+	}
+	const auto is_valid_value_bool = parse_bool_setting(valid_value);
+	if (is_valid_value_bool) {
+		if (*is_valid_value_bool == true) {
+			is_positive_bool_valid = true;
+		} else {
+			is_negative_bool_valid = true;
+		}
```

Comment:
Set the other to false, as a precaution?

Reply:
They're separate (and not connected); they record if the property access positive bools and negative bools. For example:
 - `sbtype` only allows negatives but not positives, but
 - `composite` allows both positives and negatives